wsmaster/codenvy-ldap/src/main/java/com/codenvy/ldap/sync/ProfileMapper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.ldap.sync;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.eclipse.che.api.user.server.model.impl.ProfileImpl;
+import org.eclipse.che.commons.lang.Pair;
+import org.ldaptive.LdapAttribute;
+import org.ldaptive.LdapEntry;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * @author Yevhenii Voevodin
+ */
+public class ProfileMapper implements Function<LdapEntry, ProfileImpl> {
+
+    /** App attribute name -> ldap attribute name . */
+    private final ImmutableMap<String, String> appToLdapAttrNames;
```

Comment:
The map won't be mutable in runtime which is clear from the context, so i think you question is related rather to the declaration, my answer is that it is more obvious that map is immutable and less error prone for continuous development, while `Map` interface doesn't say anything about mutability, so as for me it is additional important information for developer